bubble_sort_by.rb

```diff
@@ -0,0 +1,25 @@
+# creating bubble sort that accept a block
+
+def bubble_sort_by(arr)
+  arr.each do |_left, _right|
+    i = 0
+    swap = 0
+
+    while i < arr.length - 1
+      yield_ = yield(arr[i], arr[i + 1]) if block_given?
+
+      yield_ = arr[i] > arr[i + 1]
+      if yield_
+        temp = arr[i]
+        arr[i] = arr[i + 1]
+        arr[i + 1] = temp
+      end
+      i += 1
+      swap += 1
+    end
+    break if swap.zero?
+  end
+  arr
+end
```

Comment:
- The same issue as mentioned by @whiz25 except that the method is currently returning `["hello", "hey", "hi"]`.

  - the `yield_` assignments lines in your method always take the second assignment as a final value, you should distinguish between the case where a block is given and the case where it's not.